### PR TITLE
add new demo notebook for louvain

### PR DESCRIPTION
**Pull request by ChuckHastings**
Added a demonstration notebook for running MNMG Louvain

### CODE REVIEW

File: notebooks/demo/mg_louvain.ipynb [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+    "### Basic setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import needed libraries. We recommend using cugraph_dev env through conda\n",
```
Comment (rlratzel) [inline notebooks/demo/mg_louvain.ipynb:32]:
```suggestion
    "# Import needed libraries. We recommend using the [cugraph_dev](https://github.com/rapidsai/cugraph/tree/branch-21.12/conda/environments) env through conda\n",
```

File: notebooks/demo/mg_louvain.ipynb [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The Hollywood dataset is in our S3 bucket and zipped.  \n",
+    "1. We'll need to create a folder for our data in the `/data` folder\n",
+    "1. Download the zipped data into that folder from S3 (it will take some time as it it 6GB)\n",
+    "1. Decompress the zipped data for use (it will take some time as it it 26GB)"
```
Comment (rlratzel) [inline notebooks/demo/mg_louvain.ipynb:56]: Do these get rendered properly in jupyter or do they all show up as `1.` as seen here?

File: notebooks/demo/mg_louvain.ipynb [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+    }
+   ],
+   "source": [
+    "import urllib.request\n",
+    "import os\n",
+    "\n",
+    "data_dir = '../data/'\n",
+    "if not os.path.exists(data_dir):\n",
+    "    print('creating data directory')\n",
+    "    os.system('mkdir ../data')\n",
```
Comment (rlratzel) [inline notebooks/demo/mg_louvain.ipynb:81]: A more modern way would be to use `pathlib`:
```
import pathlib
pathlib.Path("../data").mkdir()
```
but then you'd be better off updating all the path operations to use `pathlib` for consistency, so if you'd rather stick with the `os` module already imported:
```
os.mkdir("../data")
```

File: notebooks/demo/mg_louvain.ipynb [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+     "text": [
+      "Louvain:  8.262660264968872 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Start Pagerank timer\n",
+    "t_start = time.time()\n",
+    "\n",
+    "# Get the pagerank scores\n",
```
Comment (rlratzel) [inline notebooks/demo/mg_louvain.ipynb:214]:
```suggestion
    "# Get the Louvain partition assignments for each vertex and the global modularity score.\n",
```

**Review by rlratzel — CHANGES_REQUESTED** [review]

File: notebooks/demo/mg_louvain.ipynb (hunk 3 above)
Comment (ChuckHastings) [reply notebooks/demo/mg_louvain.ipynb:81]: We had briefly discussed in slack the idea of making a set of utility functions to support this in a more general way (retrieving test data).  Would you like me to update this, or do you want to defer until we have a better tool?

File: notebooks/demo/mg_louvain.ipynb (hunk 2 above)
Comment (ChuckHastings) [reply notebooks/demo/mg_louvain.ipynb:56]: They get rendered properly in jupyter.

**Comment by ChuckHastings** [comment]
rerun tests

File: notebooks/demo/mg_louvain.ipynb (hunk 3 above)
Comment (rlratzel) [reply notebooks/demo/mg_louvain.ipynb:81]: Ah, thanks for the reminder.  I've added [this issue](https://github.com/rapidsai/cugraph/issues/1935) to track it, so feel free to either implement it if you'd like or ignore it in favor of a future utility call.

**Comment by ChuckHastings** [comment]
@gpucibot merge

**Pull request merged**